packages/material-react-table/src/components/menus/MRT_ShowHideColumnsMenu.tsx

```diff
@@ -45,12 +44,6 @@ export const MRT_ShowHideColumnsMenu = <TData extends MRT_RowData>({
   } = table;
   const { columnOrder, columnPinning, density } = getState();
 
-  const handleToggleAllColumns = (value?: boolean) => {
```

Comment:
I remember having good reasons for doing it this way. I think it had to do with hiding and showing grouped columns correctly. Will need to test further. At worst, may have to just use a `table.setColumnVisibility(prev => /**/)` call with the correct ids

Reply:
You are correct!

Hiding all columns in grouped scenario now looks as follow:
![image](https://github.com/user-attachments/assets/91095855-65a8-423d-9f34-8abc4cb62e20)


MIght dig a bit deeper to find the root cause

Reply:
When hiding all columns within the first grouping example the output is the same as on the official documentation:

![image](https://github.com/user-attachments/assets/fa914fff-da24-484a-a72d-d4f622f52b3f)

But please test further to find some egde cases